src/iuse_actor.cpp

```diff
@@ -2583,6 +2599,16 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
     if( action == RT_REPAIR ) {
         if( roll == SUCCESS ) {
             pl.add_msg_if_player(m_good, _("You repair your %s!"), fix.tname().c_str());
+            if( t->has_quality( quality_sew ) ) {
+                p->add_msg_if_player(m_info,
+                    _("You have used %d filament charges."), filament_required);
+                for (const auto &it : req.get_components()) {
+                    p->consume_items(it);
+                }
+                for (const auto &it : req.get_tools()) {
+                    p->consume_tools(it);
+                }
```

Comment:
Thanks, thread charges are really spent each repair tick - I will search for a solution to spent `thread` each repair tick without thread charges in sewing tools.

Soldering is not affected as it is has somewhat different mechanics:

**master**:

* `soldering iron` has `battery` charges which are spent on each repair tick and `plastic`/`kevlar`/`metal` is only used when repair succeeded;
* `sewing kit` has `thread` charges which are spent each tick and `rag`/`leather`/`fur`/`nomex` is only used when repair succeeded.

**this PR int its current state**:

* Nothing has been changed - `soldering iron` still has `battery` charges which are spent on each repair tick and `plastic`/`kevlar`/`metal` is only used when repair succeeded;
* `sewing kit` has no charges to spend each tick, `thread` and `rag`/`leather`/`fur`/`nomex` are both considered material requirements and used only when repair succeeded.